usr.sbin/bhyve/block_if.h

```diff
@@ -75,4 +138,55 @@ int	blockif_delete(struct blockif_ctxt *bc, struct blockif_req *breq);
 int	blockif_cancel(struct blockif_ctxt *bc, struct blockif_req *breq);
 int	blockif_close(struct blockif_ctxt *bc);
 
+/*
+ * Each block device backend registers a set of function pointers that are
+ * used to implement the net backends API.
+ */
+struct block_backend {
+	const char *bb_name;		/* identifier used parse the option string */
+	/*
+	 * Routines used to initialize and cleanup the resources needed
+	 * by a backend. The init and cleanup function are used internally,
+	 * and should not be called by the frontend.
+	 */
+	void	(*bb_init)(void);
+
+	void	(*bb_cleanup)(blockif_ctxt_t *bc);
+
+	blockif_ctxt_t * (*bb_open)(const char *optstr, const char *ident);
+
+        off_t	(*bb_size)(blockif_ctxt_t *bc);
+
+        void	(*bb_chs)(blockif_ctxt_t *bc, uint16_t *c, uint8_t *h, 
+ 		       uint8_t *s);
+
+	int	(*bb_sectsz)(blockif_ctxt_t *bc);
+
+        void	(*bb_psectsz)(blockif_ctxt_t *bc, int *size, int *off);
+
+        int	(*bb_queuesz)(blockif_ctxt_t *bc);
+
+        int	(*bb_is_ro)(blockif_ctxt_t *bc);
+
+        int	(*bb_candelete)(blockif_ctxt_t *bc);
+
+        int	(*bb_read)(blockif_ctxt_t *bc, blockif_req_t *breq);
+
+        int	(*bb_write)(blockif_ctxt_t *bc, blockif_req_t *breq);
+
+        int	(*bb_flush)(blockif_ctxt_t *bc, blockif_req_t *breq);
+
+        int	(*bb_delete)(blockif_ctxt_t *bc, blockif_req_t *breq);
+
+        int	(*bb_cancel)(blockif_ctxt_t *bc, blockif_req_t *breq);
+
+        int	(*bb_close)(blockif_ctxt_t *bc);
+
+	/* Room for backend-specific data. */
+	void *bb_opaque;
+};
```

Comment:
If we are going to have backends outside of the base code, I think we should have a header file with this structure defined somewhere in /usr/include.

Reply:
I think that that is going to be a good fix for not having to include everything and the kitchensick in the port.
something like `/usr/include/bhyve/block_if.h` ??

Reply:
yes